### DISCUSSION TITLE
Help with AWS cli

### DISCUSSION THREAD

**Discussion by mohammed-salam** [community/community#26398] [Actions]
<p>I been trying to use the AWS cli with github actions. I’m using this workflow <a href="https://github.com/actions/aws/tree/master/cli" rel="nofollow noopener">https://github.com/actions/aws/tree/master/cli</a> to run an aws s3 sync command. Does anyone have any working examples? Here’s an example of what I been trying:</p>
<p>jobs:<br>
  library-update:<br>
    runs-on: ubuntu-latest<br>
    steps:<br>
      - uses: actions/aws/cli@master<br>
      - with:<br>
        - args: s3 sync location1 location2 --acl public-read<br>
        - secrets:<br>
          - AWS_ACCESS_KEY_ID<br>
          - AWS_SECRET_ACCESS_KEY</p>

**Comment by jherico** [comment, marked as answer]
<p>Personally I would recommend using a python script instead of the AWS CLI action, since the latter is only supported on Linux, and it’s documentation apparently hasn’t been fully updated to reflect the YAML syntax.  In fact the last commit was <em>removing</em> the example YAML syntax.  </p>
<p>I use the boto3 module for Python, which you can get like so</p>
<p><span>    - </span><span>name</span><span>: </span><span>Install python modules</span><br>
<span>      </span><span>run</span><span>: </span><span>pip install boto3</span></p>
<p>I then folow up with running a script something like this:</p>
<p><span>    - </span><span>name</span><span>: </span><span>Upload artifact</span><br>
<span>      </span><span>working-directory</span><span>: </span><span>${{runner.workspace}}/build</span><span>  </span></p>
<p><span>      </span><span>shell</span><span>: </span><span>bash</span><span>      </span><span>env</span><span>:  </span><br>
<span>        </span><span>AWS_ACCESS_KEY_ID</span><span>: </span><span>${{ secrets.aws_access_key_id }}</span><br>
<span>        </span><span>AWS_SECRET_ACCESS_KEY</span><span>: </span><span>${{ secrets.aws_secret_access_key }}</span><span> </span><br>
<span>        </span><span>ARTIFACT_PATTERN</span><span>: </span><span>HighFidelity-Beta-*.exe</span><br>
<span>      </span><span>run</span><span>: </span><span>python “$GITHUB_WORKSPACE\tools\ci-scripts\upload.py”</span><span> </span></p>
<p>Specifying bash for the shell means you can use the $ syntax for environment variables regardless of platform.</p>
<p>Where the script contents look like this:</p>
<p><span>import</span><span> os</span><br>
<span>import</span><span> boto3</span><br>
<span>import</span><span> glob</span></p>
<p><span>bucket_name </span><span>=</span><span> os.environ[</span><span>‘BUCKET_NAME’</span><span>]</span><br>
<span>upload_prefix </span><span>=</span><span> os.environ[</span><span>‘UPLOAD_PREFIX’</span><span>]</span><br>
<span>S3 </span><span>=</span><span> boto3.client(</span><span>‘s3’</span><span>)</span></p>
<p><span>path </span><span>=</span><span> os.path.join(os.getcwd(), os.environ[</span><span>‘ARTIFACT_PATTERN’</span><span>])</span><br>
<span>files </span><span>=</span><span> glob.glob(path, </span><span>recursive</span><span>=</span><span>False</span><span>)</span><br>
<span>for</span><span> archiveFile </span><span>in</span><span> files:</span><br>
<span>    filePath, fileName </span><span>=</span><span> os.path.split(archiveFile)</span><br>
<span>    S3.upload_file(os.path.join(filePath, fileName), bucket_name, upload_prefix </span><span>+</span><span> </span><span>‘/’</span><span> </span><span>+</span><span> fileName)</span><br>
This ends up being more portable IMO, although on OSX builds you have to use <code>pip3</code> and <code>python3</code> instead of <code>pip</code> and <code>python</code></p>
<p>This <em>does</em> pre-suppose that you’ve done a checkout of some repository containing your script, otherwise you’d have to find a way to express the python logic directly in the workflow.  Also it looks like boto3 doesn’t have a <code>sync</code> command, so you might have to reproduce the logic.</p>

**Comment by jungheelee** [comment]
<p>AWS officially provide github actions </p>
<p><a href="https://github.com/aws-actions/configure-aws-credentials" rel="nofollow noopener">https://github.com/aws-actions/configure-aws-credentials</a></p>
<ol>
<li>
<p>get your AWS_ACCESS_KEY_ID &amp; AWS_SECRET_ACCESS_KEY (on AWS, IAM user)</p>
</li>
<li>
<p>set your access &amp; secret on github</p>
</li>
</ol>
<ul>
<li>repository &gt; setting &gt; Secrets </li>
<li>add a new secret </li>
<li>example name is <code>AWS_ACCESS_KEY_ID_MAY</code> and <code>AWS_SECRET_ACCESS_KEY_MAY</code>
</li>
</ul>
<ol start="3">
<li>using on workflow</li>
</ol>
<p>${{ secrets.YOUR_SECRETS_NAME }}</p>
<p>example</p>
<pre><code>-name: Configure AWS Credentials
uses: aws-actions/configure-aws-credentials@v1.0.1
with:
aws-access-key-id: ${{ secrets. **AWS\_ACCESS\_KEY\_ID\_MAY** }}
aws-secret-access-key: ${{ secrets. **AWS\_SECRET\_ACCESS\_KEY\_MAY** }}
aws-region: ap-northeast-2# aws seoul region 
</code></pre>

**Comment by unfor19** [comment]
<p><a class="mention" href="/mohammed-salam">@mohammed-salam</a>, if it’s still relevant …</p>
<p>I’ve created a GitHub Action that installs the AWS CLI on a Linux runner, according to a given version, so you might find it useful - <a href="https://github.com/marketplace/actions/install-aws-cli-action" rel="noopener nofollow ugc">unfor19/install-aws-cli-action</a></p>
<p>This is how you use it -</p>
<pre><code class="lang-yaml">- id: install-aws-cli
  uses: unfor19/install-aws-cli-action@v1
  with:
    version: 1
</code></pre>
